wagtail/core/models.py

```diff
@@ -2839,21 +2839,23 @@ def workflow_in_progress(self):
     def current_workflow_state(self):
         """Returns the in progress or needs changes workflow state on this page, if it exists"""
         try:
-            return WorkflowState.objects.active().get(page=self)
+            return WorkflowState.objects.active().select_related("current_task_state__task").get(page=self)
         except WorkflowState.DoesNotExist:
             return
 
     @property
     def current_workflow_task_state(self):
         """Returns (specific class of) the current task state of the workflow on this page, if it exists"""
-        if self.current_workflow_state and self.current_workflow_state.status == WorkflowState.STATUS_IN_PROGRESS and self.current_workflow_state.current_task_state:
-            return self.current_workflow_state.current_task_state.specific
+        current_workflow_state = self.current_workflow_state
```

Comment:
I'm guessing a huge share of the difference in queries comes from this one line, would be interesting to test with and without it to verify.

Reply:
@femto113 Yes, if we fix only this  method number goes down from 400 to 200. 
also as you saw PR has two commits: 

1. first commit reduces number db queries to 90
2. second commit reduces to 60

Let me know which variant do you prefer and I will update PR